api/resource/resource.go

```diff
@@ -523,6 +523,9 @@ func mergeStringMapsWithBuildAnnotations(maps ...map[string]string) map[string]s
 	result := mergeStringMaps(maps...)
 	for i := range BuildAnnotations {
 		if len(maps) > 0 {
+			if BuildAnnotations[i] == utils.BuildAnnotationsGenAddHashSuffix {
```

Comment:
Could you add comments for why `BuildAnnotationsGenAddHashSuffix` is excepted?

Reply:
That makes sense. I added a comment there.
https://github.com/kubernetes-sigs/kustomize/pull/5117/commits/ed4d75504f68706a6a4f81200794135f74027ad2

Reply:
Thanks!